### DISCUSSION TITLE
[repo access][logs] Is it possible to surface inaccessible repos in Renovate logs?

### DISCUSSION THREAD

**Discussion by slonim-greg** [renovatebot/renovate#24064] [Suggest an Idea]
### Problem
When a token used by Renovate does not have access to repos specified in the `autodiscoverFilter`, it appears difficult to identify the access problem; logs do not state lack of access and repos are ignored.

### Context
- We run GL and use "groups" to organise projects and access to projects, e.g. `/organisation/team-A/project-A`
- Our GitLab bot's token was initially given access to the `/organisation` level, but subsequently the token expired
- Our SRE recreated the token, but by default it had access to only the `project-A` bit of above example

Therefore, the bot was only discovering repos under `project` but no others, and it was not clear from the logs why we were only seeing `starting repo A` and `finishing repo A` while we had repos X, Y, Z, etc. in the `autodiscoverFilter` which live under different team groups.
I had the token access elevated all the way to the `organisation` level, but it took a bit of digging to understand the problem.

### Ask
Would it be possible to surface this in the logs, e.g. "You specified repo X in your `autodiscoverFilter` but the bot cannot access this repository due to lack of permissions."

**Comment by rarkins** [comment]
Yes, you could print a log message for any filter which matches 0 repositories. PRs welcome

**Reply by slonim-greg** [reply]
Brilliant, will contribute. Thanks, Rhys.

**Reply by slonim-greg** [reply]
There is already a log message for a filter that matches 0 repositories [here](https://github.com/renovatebot/renovate/blob/main/lib/workers/global/autodiscover.ts#L45) and [here](https://github.com/renovatebot/renovate/blob/main/lib/workers/global/autodiscover.ts#L60).

I was interested in a use case where some configured repositories (via `repositories` or `autodiscoverFilter`) were not discovered but `autodiscoverFilter` may have edge cases where simply comparing a number of discovered with number of filter elements is not sufficient enough. I think that spending time on solving a problem of producing a log message might not be worth it in this case.
